### PR TITLE
feat(react): implement `useBlockProfiles`

### PR DESCRIPTION
**Pull request by reecejohnson**

### CODE REVIEW

File: packages/react/src/transactions/adapters/PendingTransactionGateway/PendingTransactionGateway.ts [hunk 1]
```diff
@@ -43,6 +43,7 @@ const lensHubTransactionKinds = [
 
 const transactionKindToFilterGroup: { [k in TransactionKind]: TransactionKind[] } = {
   [TransactionKind.ACT_ON_PUBLICATION]: lensHubTransactionKinds,
+  [TransactionKind.BLOCK_PROFILE]: lensHubTransactionKinds,
```
Comment (cesarenaldi) [inline packages/react/src/transactions/adapters/PendingTransactionGateway/PendingTransactionGateway.ts:46]: did you check if this exposed on the hub?

File: packages/react/src/transactions/adapters/profiles/BlockProfilesGateway.ts [hunk 2]
```diff
@@ -0,0 +1,130 @@
...
+    private readonly transactionFactory: ITransactionFactory<BlockProfilesRequest>,
+  ) {}
+
+  async createDelegatedTransaction(
+    request: BlockProfilesRequest,
+  ): PromiseResult<Transaction<BlockProfilesRequest>, BroadcastingError> {
+    const result = await this.broadcast(request);
+
+    if (result.isFailure()) {
+      return failure(result.error);
```
Comment (cesarenaldi) [inline packages/react/src/transactions/adapters/profiles/BlockProfilesGateway.ts:44]: nit: 
```suggestion
      return result
```

File: packages/react/src/transactions/adapters/responders/BlockProfilesResponder.ts [hunk 3]
```diff
@@ -0,0 +1,40 @@
...
+          ...current.operations,
+          canBlock: false,
+          isBlockedByMe: {
+            __typename: 'OptimisticStatusResult',
+            isFinalisedOnchain: false,
+            value: true,
+          },
+        },
+      }));
+    });
```
Comment (cesarenaldi) [inline packages/react/src/transactions/adapters/responders/BlockProfilesResponder.ts:26]: Any plan to move this to field policy?

File: packages/react/src/transactions/adapters/useBlockProfileController.ts [hunk 4]
```diff
@@ -0,0 +1,62 @@
...
+      transactionQueue,
+      presenter,
+    );
+
+    await blockProfile.execute(request);
+
+    const result = presenter.asResult();
+
+    if (result.isSuccess()) {
+      return result.value.waitForCompletion();
```
Comment (cesarenaldi) [inline packages/react/src/transactions/adapters/useBlockProfileController.ts:57]: do we need to wait for completion here?

File: packages/react/src/transactions/adapters/profiles/BlockProfilesGateway.ts (hunk 2 above)
Comment (reecejohnson) [reply packages/react/src/transactions/adapters/profiles/BlockProfilesGateway.ts:44]: thanks

**Comment by reecejohnson** [comment]
closing - merged with #638

**Pull request closed without merging**